core/src/main/scala/org/apache/spark/ErrorInfo.scala

```diff
@@ -82,11 +85,11 @@ private[spark] object SparkThrowableHelper {
       val subMessageParameters = messageParameters.tail
       "[" + errorClass + "." + subErrorClass + "] " + String.format((errorInfo.messageFormat +
         errorSubInfo.messageFormat).replaceAll("<[a-zA-Z0-9_-]+>", "%s"),
-        subMessageParameters: _*)
+        subMessageParameters: _*) + queryContext
```

Comment:
I know it's causing diffs, but we really should have a space.

```suggestion
        subMessageParameters: _*) + " " + queryContext
```

Reply:
Actually, there is a "\n" at the beginning of `queryContext`.
Probably I should move it from `queryContext` to the method here.

Reply:
I tried moving the "\n" out of the query context. It requires updates on the starting and ending position of "^^^^^^" too. I will do it in another PR to make this one clean.

Reply:
Opened https://github.com/apache/spark/pull/36612 for this.